### PR TITLE
Add CrowdSec TIProvider

### PR DESCRIPTION
**Pull request by sbs2001**
Added CrowdSec TIProvider. 

Unit tests and documentation changes included.

### CODE REVIEW

**Review by ianhelle — COMMENTED** [review]
Looks good to me although Pete said he had spotted an item in the unit test.
I'll leave for him to add comments

**Comment by ianhelle** [comment]
Looks like you have an import error (from the tests).
Very likely `import datetime` vs `from datetime import datetime`
```
tests/context/test_tiproviders.py:935: in <module>
    "first_seen": f"{dt.now().isoformat(timespec='seconds')}+00:00",
E   AttributeError: module 'datetime' has no attribute 'now'
```

File: docs/source/data_acquisition/TIProviders.rst [hunk 1]
```diff
@@ -268,6 +269,11 @@ fictitious - the format of the keys may differ from what is shown
           TenantID: 228d7b5f-4920-4f8e-872f-52072b92b651
         Primary: True
         Provider: "AzSTI"
+      CrowdSec:
+        Args:
+          AuthKey: 12ffVvqTaV1fYf7eieoic21rftntHQhO2Hdsxyzabcd
```
[the remark this answers is not in the document]
Comment (sbs2001) [reply docs/source/data_acquisition/TIProviders.rst]: Done

File: docs/source/getting_started/msticpyconfig.rst [hunk 2]
```diff
@@ -136,6 +137,11 @@ Currently supported provider names are:
             KeyVault:
         Primary: False
         Provider: "OPR"
+      CrowdSec:
+        Args:
+          AuthKey: 12ffVvqTaV1fYf7eieoic21rftntHQhO2Hdsxyzabcd
```
[the remark this answers is not in the document]
Comment (sbs2001) [reply docs/source/getting_started/msticpyconfig.rst]: Done

File: msticpy/context/tiproviders/crowdsec.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    _QUERIES = {
+        "ipv4": APILookupParams(
+            path="/v2/smoke/{observable}",
+            headers={
+                "x-api-key": "{AuthKey}",
+                "User-Agent": "crowdsec-msticpy-tiprovider/v1.0.0",
+            },
+        ),
+        "ipv6": APILookupParams(
```
[the remark this answers is not in the document]
Comment (sbs2001) [reply msticpy/context/tiproviders/crowdsec.py]: Thanks, I've done it now.

**Comment by sbs2001** [comment]
@ianhelle Thanks for the review ! I've pushed commits to implement your suggestions.

**Review by ianhelle — APPROVED** [review]

**Pull request merged**